lib/internal/Magento/Framework/App/Config/ScopeCodeResolver.php

```diff
@@ -34,28 +35,32 @@ public function __construct(ScopeResolverPool $scopeResolverPool)
      * Resolve scope code
      *
      * @param string $scopeType
-     * @param string $scopeCode
+     * @param string|null $scopeCode
      * @return string
      */
     public function resolve($scopeType, $scopeCode)
     {
         if (isset($this->resolvedScopeCodes[$scopeType][$scopeCode])) {
             return $this->resolvedScopeCodes[$scopeType][$scopeCode];
         }
-        if (($scopeCode === null || is_numeric($scopeCode))
-            && $scopeType !== ScopeConfigInterface::SCOPE_TYPE_DEFAULT
-        ) {
+
+        if ($scopeType !== ScopeConfigInterface::SCOPE_TYPE_DEFAULT) {
             $scopeResolver = $this->scopeResolverPool->get($scopeType);
             $resolverScopeCode = $scopeResolver->getScope($scopeCode);
         } else {
             $resolverScopeCode = $scopeCode;
         }
 
-        if ($resolverScopeCode instanceof \Magento\Framework\App\ScopeInterface) {
+        if ($resolverScopeCode instanceof ScopeInterface) {
             $resolverScopeCode = $resolverScopeCode->getCode();
         }
 
+        if ($scopeCode == null) {
```

Comment:
I suggest changing this statement according to the general rule of preferring the identical comparison operator `===` over the equal one `==`.

Reply:
Thanks @aleron75 — confirming this has been done